opensrp-app/src/main/java/org/smartregister/sync/intent/SyncIntentService.java

```diff
@@ -79,12 +82,16 @@ private void doSync() {
             if (hasValidAuthorization || !CoreLibrary.getInstance().getSyncConfiguration().disableSyncToServerIfUserIsDisabled()) {
                 pushToServer();
             }
-            if (hasValidAuthorization) {
-                pullECFromServer();
-            } else {
+
+            if (!syncUtils.isAppVersionAllowed()) {
+                AllSettings settingsRepository = org.smartregister.Context.getInstance().allSettings();
+                settingsRepository.put(IS_APP_VERSION_ALLOWED, "false");
```

Comment:
This should be after `if(!hasValidAuthorization)` as its more efficient as `hasValidAuthorization` is already known.

Not sure why  `IS_APP_VERSION_ALLOWED` is being re-saved when its used to determine `syncUtils.isAppVersionAllowed()`

Reply:
Fixed the first concern.

I'm not sure I understand the second concern. An upgrade can only be done by uninstalling the current app or by clearing data on the phone and upgrading with a new apk. So the `IS_APP_VERSION_ALLOWED` will definitely be recomputed.

Once a user is logged out, they cannot log back in until they upgrade the app by following one of the two options above.

Reply:
Its possible to upgrade without uninstall of the current app and this is the upgrade recommended for the OpenSRP.

The concert is valid when the APK is upgraded

Reply:
Fixed.